### PR TITLE
fix: tests and styles

### PR DESCRIPTION
**Pull request by bluet**
port to python >=3.8

deprecate <3.8

### CODE REVIEW

File: proxybroker/negotiators.py [hunk 1]
```diff
@@ -61,6 +62,8 @@ async def negotiate(self, **kwargs):
         await self._proxy.send(struct.pack('3B', 5, 1, 0))
         resp = await self._proxy.recv(2)
 
+        if not isinstance(resp, (bytes, str)):
```
Comment (afunTW) [inline proxybroker/negotiators.py:65]: Have you encountered any issues here?
Because the response comes from the python build-in `asyncio.StreamReader`, it should return the bytes object.

File: proxybroker/negotiators.py [hunk 2]
```diff
@@ -92,6 +95,9 @@ async def negotiate(self, **kwargs):
 
         await self._proxy.send(struct.pack('>2BH5B', 4, 1, port, *bip, 0))
         resp = await self._proxy.recv(8)
+        if isinstance(resp, asyncio.Future):
```
Comment (afunTW) [inline proxybroker/negotiators.py:98]: Have you encountered any issues here?
Because the response comes from `asyncio` and it should only possibly be the `Future` object.

File: proxybroker/utils.py [hunk 3]
```diff
@@ -58,6 +58,8 @@ def get_all_ip(page):
 
 def get_status_code(resp, start=9, stop=12):
     try:
+        if not isinstance(resp, (bytes, str)):
```
Comment (afunTW) [inline proxybroker/utils.py:61]: From the viewpoint to check the external argument is acceptable, but the resp in this module should be all bytes object, and what I would like to know is there any edge case happened?

File: tests/test_negotiators.py [hunk 4]
```diff
@@ -44,13 +44,13 @@ def test_base_attrs(proxy, ngtr, check_anon_lvl, use_full_path):
         (
             'SOCKS5',
             80,
-            future_iter(b'\x05\x00', b'\x05\x00\x00\x01\xc0\xa8\x00\x18\xce\xdf'),
```
Comment (afunTW) [inline tests/test_negotiators.py:47]: Any reason to remove this? And so does all operation in `tests` folder

**Comment by bluet** [comment]
@afunTW I was just trying to fetch the changes from @a5r0n 's branch hahaha

hi @a5r0n would you be ok to discuss here?

**Comment by a5r0n** [comment]
@bluet  i'm not really in the context, it's been a while, but let's give it a try

**Comment by afunTW** [comment]
@a5r0n I revise the CI flow in https://github.com/bluet/proxybroker2/pull/79 and https://github.com/bluet/proxybroker2/pull/80, would you like to rebase and trigger the latest CI again?

**Comment by afunTW** [comment]
Since the PR mainly revise the `asyncio` break change https://docs.python.org/3/library/asyncio-queue.html#asyncio.Queue, and has been test. Will merge for further maintainese.

**Comment by bluet** [comment]
Awesome 👍
